### DISCUSSION TITLE
Change global parameter inside a function

### DISCUSSION THREAD

**Discussion by fspinar** [jspsych/jsPsych#2086] [Q&A]
Hi all,

I have a function that determines the stimulus of a trial. That is:

var trial = {
...
stimulus: function(){

}
}

Inside this function I need to be able to update a global parameter and use that parameter in other trials. That is :
var parameter = 0;
var trial = {
...
stimulus: function(){
parameter = parameter + 1;
}
}

var newTrial = {
// access to this parameter
}

Could you help me with this ? Should I add this parameter to the data structure of the experiment ? (data.parameter = ...) 
I suppose this is the easiest way ? 

Thank you and best regards

**Comment by becky-gilbert** [comment]
Hi @fspinar, I'm moving this to the Discussions section because it's a question, rather than a bug or feature request.

**Reply by fspinar** [reply]
Thanks @becky-gilbert , I didn't realize I posted it on Issues, it belongs to the discussions section. Sorry about that.

Any suggestions for solution ?

Thanks a lot.
